File: JAVA-GEOMETRIA/Circulo.java
import java.awt.Color;
import java.awt.Graphics;

public class Circulo {
    private final double centroX;
    private final double centroY;
    private final double radio;
    private final Color borde;
    private final Color relleno;

    public Circulo(double centroX, double centroY, double radio, Color borde, Color relleno) {
        this.centroX = centroX;
        this.centroY = centroY;
        this.radio = radio;
        this.borde = borde;
        this.relleno = relleno;
    }

    public double calcularArea() {
        return Math.PI * Math.pow(radio, 2);
    }

    public double calcularPerimetro() {
        return 2 * Math.PI * radio;
    }

    public void dibujar(Graphics g) {
        // Dibuja el relleno
        g.setColor(relleno);
        g.fillOval((int)(centroX - radio), (int)(centroY - radio), (int)(radio * 2), (int)(radio * 2));
        
        // Dibuja el borde
        g.setColor(borde);
        g.drawOval((int)(centroX - radio), (int)(centroY - radio), (int)(radio * 2), (int)(radio * 2));

        // Dibuja el área y el perímetro
        g.setColor(Color.BLACK);
        g.drawString(String.format("Área: %.2f", calcularArea()), (int)centroX, (int)(centroY + radio + 15));
        g.drawString(String.format("Perímetro: %.2f", calcularPerimetro()), (int)centroX, (int)(centroY + radio + 30));
    }
}

File: CSHARDP-GEOMETRIA/Circulo.cs
using System.Drawing;

namespace CSHARDP_GEOMETRIA
{
    public class Circulo
    {
        private double CentroX;
        private double CentroY;
        private double Radio;
        private Color Borde;
        private Color Relleno;

        public Circulo(double centroX, double centroY, double radio, Color borde, Color relleno)
        {
            CentroX = centroX;
            CentroY = centroY;
            Radio = radio;
            Borde = borde;
            Relleno = relleno;
        }

        public double CalcularArea()
        {
            return Math.PI * (Radio * Radio);
        }

        public double CalcularPerimetro()
        {
            return 2 * Math.PI * Radio;
        }

        public void Dibujar(Graphics g)
        {
            using (Brush brush = new SolidBrush(Relleno))
            {
                g.FillEllipse(brush, (float)(CentroX - Radio), (float)(CentroY - Radio), (float)(Radio * 2), (float)(Radio * 2));
            }
            using (Pen pen = new Pen(Borde))
            {
                g.DrawEllipse(pen, (float)(CentroX - Radio), (float)(CentroY - Radio), (float)(Radio * 2), (float)(Radio * 2));
            }

            g.DrawString($"Área: {CalcularArea():F2}", SystemFonts.DefaultFont, Brushes.Black, (float)CentroX, (float)CentroY + (float)Radio + 5);
            g.DrawString($"Perímetro: {CalcularPerimetro():F2}", SystemFonts.DefaultFont, Brushes.Black, (float)CentroX, (float)CentroY + (float)Radio + 20);
        }
    }
}
